raft/log.go

```diff
@@ -1353,17 +1355,21 @@ func (l *Log) append(e *LogEntry) error {
 	}
 
 	assert(e.Index == l.lastLogIndex+1, "log entry skipped(%d): idx=%d, prev=%d", l.id, e.Index, l.lastLogIndex)
-
-	// Encode entry to a byte slice.
-	buf := make([]byte, logEntryHeaderSize+len(e.Data))
-	copy(buf, e.encodedHeader())
-	copy(buf[logEntryHeaderSize:], e.Data)
+	if len(l.entries) > 0 {
```

Comment:
Yes, this is to ensure that `Log.entries` is always contiguous. A similar check used to exist but was altered with some changes around `Log.lastLogIndex`